### DISCUSSION TITLE
Calculating effective mode areas with meep/MPG

### DISCUSSION THREAD

**Discussion by Edward-Deacon** [gdsfactory/gdsfactory#1185] [Q&A] [answered]
Hi everyone, I am using the MPB mode solver in gdsfactory and want to determine the effective mode areas and TE/TM fraction for modes calculated for stip waveguides. I am using the `gm.find_modes_waveguide` function to calculate the modes and I can see the `Mode `class that is returned has the properties `fraction_te, fraction_tm and effective_area` though these don't seem to be calculated when running the function. How should I go about calculating these properties? Any help would be much appreciated

**Comment by joamatab** [comment, marked as answer]
Hi Edward,

I've been using more femwell and tidy3d mode solver than MPB

You can read the [Femwell](https://helgegehring.github.io/femwell/intro.html) that can already provide you with fraction_te
For mode effective area you can see how it's done in this package
https://modes.readthedocs.io/en/latest/README.html

**Reply by Edward-Deacon** [reply]
Thanks Joaquin. I have since been able to calculate A_eff though I found there are a number of definitions for it which seem to be used. https://opg.optica.org/ol/abstract.cfm?uri=ol-37-12-2295 this paper discusses some of the definitions - the one I was looking for was equation (3) in the paper

**Reply by joamatab** [reply]
Hi Edward, 

feel free to contribute it as a merge request :)
